nixos/modules/services/networking/charybdis.nix

```diff
@@ -51,46 +51,55 @@ in
         '';
       };
 
+      motd = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        description = "Charybdis MOTD.";
+      };
+
     };
 
   };
 
 
   ###### implementation
 
-  config = mkIf cfg.enable {
-
-    users.extraUsers = singleton {
-      name = cfg.user;
-      description = "Charybdis IRC daemon user";
-      uid = config.ids.uids.ircd;
-      group = cfg.group;
-    };
-
-    users.extraGroups = singleton {
-      name = cfg.group;
-      gid = config.ids.gids.ircd;
-    };
-
-    systemd.services.charybdis = {
-      description = "Charybdis IRC daemon";
-      wantedBy = [ "multi-user.target" ];
-      environment = {
-        BANDB_DBPATH = "${cfg.statedir}/ban.db";
+  config = mkIf cfg.enable (lib.mkMerge [
+    {
+      users.extraUsers = singleton {
+        name = cfg.user;
+        description = "Charybdis IRC daemon user";
+        uid = config.ids.uids.ircd;
+        group = cfg.group;
       };
-      serviceConfig = {
-        ExecStart   = "${charybdis}/bin/charybdis-ircd -foreground -logfile /dev/stdout -configfile ${configFile}";
-        Group = cfg.group;
-        User = cfg.user;
-        PermissionsStartOnly = true; # preStart needs to run with root permissions
-      };
-      preStart = ''
-        ${coreutils}/bin/mkdir -p ${cfg.statedir}
-        ${coreutils}/bin/chown ${cfg.user}:${cfg.group} ${cfg.statedir}
-      '';
 
-    };
+      users.extraGroups = singleton {
+        name = cfg.group;
+        gid = config.ids.gids.ircd;
+      };
 
-  };
+      systemd.services.charybdis = {
+        description = "Charybdis IRC daemon";
+        wantedBy = [ "multi-user.target" ];
+        environment = {
+          BANDB_DBPATH = "${cfg.statedir}/ban.db";
+        };
+        serviceConfig = {
+          ExecStart   = "${charybdis}/bin/charybdis-ircd -foreground -logfile /dev/stdout -configfile ${configFile}";
```

Comment:
This doesn't really seem to be documented, but diving into the source shows that yes, charybdis supports this - to be precise, it will reload (only) ircd.conf on SIGHUP, (only) MOTD on SIGUSR1 and (only) ban files on SIGUSR2. Splitting it up like this seems kinda silly to me, but there it is.
I don't know how one would set up sending of those signals in the nix config for the service, though to do it for other files than MOTD would require also symlinking those to somewhere they can be changed (probably /etc/charybdis, next to the motd).